lib/spack/spack/util/compression.py

```diff
@@ -77,8 +77,12 @@ def _system_untar(archive_file):
         archive_file (str): absolute path to the archive to be extracted.
         Can be one of .tar(.[gz|bz2|xz|Z]) or .(tgz|tbz|tbz2|txz).
     """
-    outfile = os.path.basename(strip_extension(archive_file, "tar"))
-
+    archive_file_no_ext = strip_extension(archive_file)
```

Comment:
I think this should still be `strip_extension(archive_file, "tar")`: while this won't strip arbitrary extensions like ".foo", it might still do the wrong thing if handed a file in an incorrect state (e.g. a `tar.gz`), and it would be confusing if we stripped off the `.gz`, then complained about not being able to untar something with a name ending in ".tar".

Likewise for the other calls to `strip_extension`.

Reply:
This was done initially, but I run into troubles with '*.tgz', which was not removed, so the following if was evaluated true. The file was judged a file with no extension, so I ended up with an archive file <name>.tgz.tar which was incorrect.
I think your example (*.tar.gz) would work fine. gz would be stripped, so the if statement would be false. The file won't be judged as having no extension.